### PR TITLE
[SYCL] Use uniform group instructions in headers

### PR DESCRIPTION
**Pull request by aobolensk**

### CODE REVIEW

**Comment by KornevNikita** [comment]
There are changes in extension vendor & operations name. @MrSidims will we promote it as INTEL extension now and then change it to KHR?

**Comment by MrSidims** [comment]
@aobolensk please update the instructions' names to the names used here: https://github.com/KhronosGroup/SPIRV-Registry/pull/138

File: llvm-spirv/include/LLVMSPIRVExtensions.inc [hunk 1]
```diff
@@ -49,3 +49,4 @@ EXT(SPV_INTEL_bfloat16_conversion)
 EXT(SPV_INTEL_joint_matrix)
 EXT(SPV_INTEL_hw_thread_queries)
 EXT(SPV_INTEL_global_variable_decorations)
+EXT(SPV_KHR_uniform_group_instructions)
```
Comment (KornevNikita) [inline llvm-spirv/include/LLVMSPIRVExtensions.inc:52]: I don't sure if we need it here, until the extension is ready and transfered to this repository: https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1253. The same about Clang.cpp. @MrSidims WDYT?

Comment (MrSidims) [reply llvm-spirv/include/LLVMSPIRVExtensions.inc:52]: @aobolensk please remove this change from the translator, let the translator's work be merged and pulled to intel/llvm first.

File: clang/lib/Driver/ToolChains/Clang.cpp [hunk 2]
```diff
@@ -9119,7 +9119,8 @@ void SPIRVTranslator::ConstructJob(Compilation &C, const JobAction &JA,
     std::string DefaultExtArg =
         ",+SPV_EXT_shader_atomic_float_add,+SPV_EXT_shader_atomic_float_min_max"
         ",+SPV_KHR_no_integer_wrap_decoration,+SPV_KHR_float_controls"
-        ",+SPV_KHR_expect_assume,+SPV_KHR_linkonce_odr";
+        ",+SPV_KHR_expect_assume,+SPV_KHR_linkonce_odr"
+        ",+SPV_KHR_uniform_group_instructions";
```
Comment (MrSidims) [inline clang/lib/Driver/ToolChains/Clang.cpp:9123]: Let's enable the extension on any target but FPGA H/W aka move this code to line 9154.

**Review by MrSidims — CHANGES_REQUESTED** [review]

**Comment by KornevNikita** [comment]
@aobolensk https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1253 is in intel/llvm now

**Comment by aobolensk** [comment]
> @aobolensk [KhronosGroup/SPIRV-LLVM-Translator#1253](https://github.com/KhronosGroup/SPIRV-LLVM-Translator/pull/1253) is in intel/llvm now

Thanks! Rebased this on top of merged change

**Review by alexbatashev — APPROVED** [review]
Runtime LGTM

**Review by MrSidims — APPROVED** [review]

File: clang/lib/Sema/SPIRVBuiltins.td [hunk 3]
```diff
@@ -929,13 +929,13 @@ foreach name = ["GroupBroadcast"] in {
   }
 }
 
-foreach name = ["GroupIAdd", "GroupNonUniformIMul", "GroupNonUniformBitwiseOr",
-                "GroupNonUniformBitwiseXor", "GroupNonUniformBitwiseAnd"] in {
+foreach name = ["GroupIAdd", "GroupIMulKHR", "GroupBitwiseOrKHR",
+                "GroupBitwiseXorKHR", "GroupBitwiseAndKHR"] in {
```
Comment (Fznamznon) [inline clang/lib/Sema/SPIRVBuiltins.td:933]: https://github.com/KhronosGroup/SPIRV-Registry/pull/138/files?short_path=771b867#diff-771b867fa2b28c0a331ac547cfe35642695f0d4da40653e89278a2f188eb2eb4 doesn't seem to say that it removes old instructions, it says it adds new ones. So, I wonder why old ones are not preserved?

Comment (aobolensk) [reply clang/lib/Sema/SPIRVBuiltins.td:933]: Currently we're having mixed implementations in functional.hpp (uniform and non-uniform) but we want to unify it and make all of them uniform.

**Review by Fznamznon — APPROVED** [review]
FE changes look ok.

**Review by smanna12 — APPROVED** [review]
FE changes look good to me!

**Review by hchilama — APPROVED** [review]
Driver files look ok.

**Review by vladimirlaz — APPROVED** [review]

**Pull request merged**